requests/utils.py

```diff
@@ -732,6 +732,22 @@ def to_native_string(string, encoding='ascii'):
 
     return out
 
+def validate_header(header):
+    """Verifies that header value doesn't contain leading whitespace or
+    return characters. This prevents unintended header injection.
+
+    header is a tuple, in the format (name, value)
+    """
+    name, value = header
+
+    if isinstance(value, bytes):
+        pat = re.compile(b'^\S[^\\r\\n]*$|^$')
+    else:
+        pat = re.compile(r'^\S[^\r\n]*$|^$')
+    if not pat.match(value):
+        raise InvalidHeader("Invalid return character or leading space in header: %s" % name)
+
+    return (to_native_string(name), value)
```

Comment:
Frankly, I don't love that this hides the call to `to_native_string`. I understand why it's here, but I'd rather that it didn't. Can we just turn this into a function that we can pass to `filter`?